tests/install_upgrade_operators/crypto_policy/utils.py

```
@@ -261,9 +269,144 @@ def update_apiserver_crypto_policy(
+        worker_exec: ExecCommandOnPod instance for running commands on a worker node.
+        services: List of Service resources to check.
+        pqc_groups: List of PQC group names to try (e.g. ["X25519MLKEM768", "SecP256r1MLKEM768"]).
+
+    Returns:
+        dict[str, bool | None]: Mapping of service name to PQC status:
+            True = accepted PQC, False = rejected PQC, None = unreachable.
+    """
+    results: dict[str, bool | None] = {}
+    for service in services:
+        service_name = service.instance.metadata.name
+        LOGGER.info(f"Probing PQC on service: {service_name}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)